src/music/allegro_m.cpp

```diff
@@ -58,11 +59,18 @@ void MusicDriver_Allegro::Stop()
 	if (--_allegro_instance_count == 0) allegro_exit();
 }
 
-void MusicDriver_Allegro::PlaySong(const char *filename)
+void MusicDriver_Allegro::PlaySong(const MusicSongInfo &song)
 {
+	char *filename = MidiFile::GetSMFFile(song);
```

Comment:
not sure i like this change - seems a bit redundant to pass in a MusicSongInfo object if all you're going to do is look up the filename. There are less manual free's necessary in the old way - less chance for memory leaks

Reply:
The point is that with CAT format music the file may have to be decoded first, and that's what the GetSMFFile function does. I'm not sure how to handle it any better way.